### PR TITLE
feature: Added job to automate release creation

### PR DESCRIPTION
**Pull request by joaocosta-codacy**
This will also post the release notes link in the body of the release.

All that is left to do is provision the CIRCLECI_GITHUB_TOKEN with a relevant github user from our org for this to work.

### CODE REVIEW

**Review by h314to — DISMISSED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -670,7 +701,17 @@ workflows:
           force: true
           requires:
             - promote_chart_to_stable
+      - manual_release_notes_hold:
+          type: approval
+          context: CodacyDO
+          requires:
+            - tag_version_latest
```
Comment (joaocosta-codacy) [inline .circleci/config.yml:708]: I have taken the liberty of adding this final manual step to ensure that the release notes are published ahead before creating the github release with a link to their page.

File: .circleci/config.yml [hunk 2]
```diff
@@ -278,6 +278,17 @@ references:
             export EXECUTION_TIME=`date +%F_%Hh%M-%Z`
             sh /opt/tests/qa-automation-test-runner/build/run_tests.sh
 
+  append_releasenotes_link: &append_releasenotes_link
```
Comment (pauloribeiro-codacy) [inline .circleci/config.yml:281]: What does this step do? :thinking:

Comment (joaocosta-codacy) [reply .circleci/config.yml:281]: Wrongfully committed. Removing it. :)

**Comment by pauloribeiro-codacy** [comment]
Also, I just noticed that I did not place the instructions on how to (manually) add the link to the release notes on correct place in the [RELEASE.md](https://github.com/codacy/chart/blob/automate-release-notes-link/RELEASE.md) file. Right now they're inside a section to tag the CLI and the Coverage Reporter!

How would this work now for the Release Manager?

**Review by pauloribeiro-codacy — APPROVED** [review]

**Pull request merged**